libs/components/lookup/src/lib/modules/lookup/lookup.component.spec.ts

```diff
@@ -3378,6 +3406,235 @@ describe('Lookup component', function () {
+                    if (args?.context === 'modal') {
+                      return true;
+                    }
+                    const found = friends.find(
+                      (option) => option.name === anItem.name
+                    );
+                    return !found;
+                  });
+                }
+              );
+
+            component.customSearch = searchFunctionSpy;
+            // peforming a search in the popover view should provide the custom search function with a 'popover' context
+            fixture.detectChanges();
+            tick();
+
+            performSearch('s', fixture);
+            expect(searchFunctionSpy).toHaveBeenCalledWith(
+              's',
+              jasmine.arrayWithExactContents(component.data),
+              jasmine.objectContaining({
+                context: 'popover',
+              })
+            );
+          }));
+
+          describe('show more modal', () => {
+            let modalService: SkyModalService;
+
+            beforeEach(fakeAsync(() => {
+              modalService = TestBed.inject(SkyModalService);
+
+              fixture.detectChanges();
+              tick();
+            }));
+
+            // This is necessary as due to modals being launched outside of the test bed they will not
+            // automatically be disposed between tests.
+            afterEach(fakeAsync(() => {
+              // NOTE: This is important as it ensures that the modal host component is fully disposed of
+              // between tests. This is important as the modal host might need a different set of component
+              // injectors than the previous test.
+              modalService.dispose();
+              fixture.detectChanges();
+            }));
+            it('should return modal context', fakeAsync(() => {
+              const friends: any[] = component.form.controls.friends.value;
+              // creating the spy function that is pretending to be a custom search function with real functionality, because a real function is needed to be called so that some results may appear in the popover/dropdown so that the "Show more" button may be present so it can be clicked
+              searchFunctionSpy = jasmine
+                .createSpy('searchFunctionrModal')
+                .and.callFake(
+                  (
+                    searchText: string,
+                    data: any[],
+                    args?: SkyAutocompleteSearchArgs
+                  ): SkyAutocompleteSearchFunctionResponse => {
+                    return data.filter((anItem) => {
+                      if (args?.context === 'modal') {
+                        return true;
+                      }
+                      const found = friends.find(
+                        (option) => option.name === anItem.name
+                      );
+                      return !found;
+                    });
+                  }
+                );
```

Comment:
☝️ Since we're duplicating this spy multiple times in the spec, maybe we could create a new helper function to return the spy so it can be modified in one place?
```
/**
 * Some comment explaining why we need to do this...
 */
function getSearchFunctionSpy() {
  return jasmine.createSpy(...)
}

const searchFunctionSpy = getSearchFunctionSpy();

component.customSearch = searchFunctionSpy;
```

Reply:
great idea. fixed in [abstract Spy creation to helper functions](https://github.com/blackbaud/skyux/pull/381/commits/4dd35175b7154079e5b977b2e548ed1257120dda)